### PR TITLE
3501 Add dict version SavitzkyGolaySmoothd

### LINKED ISSUE #3501: Add dictionary version transform SavitzkyGolaySmoothd
**Is your feature request related to a problem? Please describe.**
Currently, we have the array version intensity transform `SavitzkyGolaySmooth`, would be nice to add the dict version.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3501 .

### Description
This PR added dict version `SavitzkyGolaySmoothd`.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_savitzky_golay_smoothd.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+class TestSavitzkyGolaySmoothd(unittest.TestCase):
+    @parameterized.expand([TEST_CASE_SINGLE_VALUE, TEST_CASE_2D_AXIS_2, TEST_CASE_SINE_SMOOTH])
+    def test_value(self, arguments, image, expected_data, atol):
+        for p in TEST_NDARRAYS:
+            result = SavitzkyGolaySmoothd(**arguments)({"img": p(image.astype(np.float32))})["img"]
+            torch.testing.assert_allclose(result, p(expected_data.astype(np.float32)), rtol=1e-4, atol=atol)
+
+
+class TestSavitzkyGolaySmoothdREP(unittest.TestCase):
+    @parameterized.expand([TEST_CASE_SINGLE_VALUE_REP])
```
Comment (wyli) [inline tests/test_savitzky_golay_smoothd.py:70]: looks like these two classes could be merged?

Comment (Nic-Ma) [reply tests/test_savitzky_golay_smoothd.py]: Hi @wyli ,

Sure, updated in the latest commit.
Could you please help review it again?

Thanks in advance.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**